smst.go

```diff
@@ -71,27 +82,39 @@ func (smst *SMST) Spec() *TrieSpec {
 	return &smst.TrieSpec
 }
 
-// Get retrieves the value digest for the given key and the digest of the value
-// along with its weight provided a leaf node exists.
+// Get retrieves the value digest for the given key, along with its weight assuming
+// the node exists, otherwise the default placeholder values are returned
 func (smst *SMST) Get(key []byte) (valueDigest []byte, weight uint64, err error) {
 	// Retrieve the value digest from the trie for the given key
-	valueDigest, err = smst.SMT.Get(key)
+	value, err := smst.SMT.Get(key)
 	if err != nil {
 		return nil, 0, err
 	}
 
-	// Check if it ias an empty branch
-	if bytes.Equal(valueDigest, defaultEmptyValue) {
+	// Check if it is an empty branch
+	if bytes.Equal(value, defaultEmptyValue) {
 		return defaultEmptyValue, 0, nil
 	}
 
+	firstSumByteIdx, firstCountByteIdx := getFirstMetaByteIdx(value)
+
+	// Extract the value digest only
+	valueDigest = value[:firstSumByteIdx]
+
 	// Retrieve the node weight
 	var weightBz [sumSizeBytes]byte
-	copy(weightBz[:], valueDigest[len(valueDigest)-sumSizeBytes:])
+	copy(weightBz[:], value[firstSumByteIdx:firstCountByteIdx])
 	weight = binary.BigEndian.Uint64(weightBz[:])
 
-	// Remove the weight from the value digest
-	valueDigest = valueDigest[:len(valueDigest)-sumSizeBytes]
+	// Retrieve the number of non-empty nodes in the sub trie
+	var countBz [countSizeBytes]byte
+	copy(countBz[:], value[firstCountByteIdx:])
+	count := binary.BigEndian.Uint64(countBz[:])
+
+	// TODO_IN_THIS_PR: Consider not retuning count here because it should always be 1
```

Comment:
```suggestion
	// TODO_IN_THIS_PR: Consider not returning count here because it should always be 1
```
IMO panicking here is a good idea - if our assumptions about the world are wrong then we need to make that clear